Add add closure variables into global scope

kai3341 opened the pull request:
Fixes #430

adamchainz commented:
I think you're off to a good start, but this is failing the tests. Please try running the tests locally with e.g. `tox -e py311`. Also, add some tests covering class closures, and a chaneglog note.

kai3341 commented:
I reproduced test fails and fixed it
Also I added new test with my code example. The most funny that these changes are work in my project but the test is failing with initial error which I try to solve
I'll continue working on this issue

kai3341 commented on src/patchy/api.py:
```diff
@@ -309,11 +309,18 @@ def _process_function() -> ast.Module:
+    globalz = dict(func.__globals__)
+
+    for cell in func.__closure__ or ():
+        cell_contents = cell.cell_contents
```
So, I was right in this question

adamchainz commented:
> Also I added new test with my code example.

I don't see this test? Or a changelog note.

adamchainz commented:
I just added two tests which fail. `func.__closure__` is `None` in the both cases. I'm not sure why it's not there...

adamchainz commented:
Rebased, extended, and merged in #620. Thanks @kai3341, got there eventually!
